examples/userapi/05_conditionaldimension.ipynb

```diff
@@ -0,0 +1,729 @@
+       "      [0., 0., 0., 1., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 1., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 1., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 1., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 0., 1.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "\n",
+    "from devito import Dimension, Function, Grid\n",
+    "import numpy as np\n",
+    "\n",
+    "# We define a 10x10 grid, by default its dimensions are x, y\n",
+    "shape = (10, 10)\n",
+    "grid = Grid(shape = shape)\n",
+    "x, y = grid.dimensions\n",
+    "\n",
+    "# Define a 2D Function 𝑓 on our grid. We will populate the data of the Function with ones on its diagonal.\n",
+    "\n",
+    "f = Function(name='f', grid=grid)\n",
+    "f.data[:, :] = np.eye(10)\n",
+    "f.data[:, :] # Print data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now lets construct a simple operator to increase the values of $ f $ by 1. We just need to update every element by 1 and execute the operator once. After the execution you can see every element updated by one."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Operator `Kernel` run in 0.01 s\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[2., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 2., 1., 1., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 2., 1., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 2., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 2., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 2., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 2., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 1., 2., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 1., 1., 2., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 1., 1., 1., 2.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "from devito import Eq, Operator\n",
+    "op = Operator(Eq(f, f + 1))\n",
+    "op.apply()\n",
+    "f.data[:,:] # Print data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Every value has been updated by one. You should be able to see 2s on the diagonal and 1s in the rest of the grid. Now let's have a look at the generated code. We can see that we iterate over a 2D space (dimensions x and y), increasing every grid value by one."
```

Comment:
```
Now let's have a look at the generated code. We can see that we iterate over a 2D space (dimensions x and y), increasing every grid value by one."
```

we can probably drop this as well as the generated code -- again, at this point they should be superfluous (@rhodrin )

Reply:
Agree.